_pytest/junitxml.py

```diff
@@ -337,6 +339,10 @@ def pytest_runtest_logreport(self, report):
                 reporter = self._opentestcase(report)
                 reporter.append_pass(report)
         elif report.failed:
+            if self.last_report and self.last_report.failed:
```

Comment:
Just to complement what @RonnyPfannschmidt said:

During normal pytest operation, report of setup/call/teardown all occur predictably because the tests are executed sequentially:

```
test1 setup
test1 call
test1 teardown
test2 setup
test2 call
test2 teardown
``` 

But using the `pytest-xdist` plugin this does not hold, because there are multiple workers running tests in parallel so the calls might end up intertwined in the `master` node, for example:

```
test1 setup
test1 call
test2 setup
test1 teardown
test2 call
test2 teardown
```